sdk/spring/spring-cloud-azure-feature-management/src/main/java/com/azure/spring/cloud/feature/management/implementation/timewindow/recurrence/RecurrenceEvaluator.java

```diff
@@ -0,0 +1,158 @@
+    /**
+     * Checks if a provided timestamp is within any recurring time window specified
+     * by the Recurrence section in the time window filter settings.
+     * @return True if the time stamp is within any recurring time window, false otherwise.
+     */
+    public static boolean isMatch(TimeWindowFilterSettings settings, ZonedDateTime now) {
+        final RecurrenceValidator validator = new RecurrenceValidator(settings);
+        if (!validator.validateSettings()) {
+            return false;
+        }
+
+        final ZonedDateTime previousOccurrence = getPreviousOccurrence(settings, now).previousOccurrence;
+        if (previousOccurrence == null) {
+            return false;
+        }
+
+        final ZonedDateTime occurrenceEndDate = previousOccurrence.plus(
+            Duration.between(settings.getStart(), settings.getEnd()));
+        return now.isBefore(occurrenceEndDate);
+    }
+
+    /**
+     * Find the most recent recurrence occurrence before the provided time stamp.
+     *
+     * @return The closest previous occurrence.
+     */
+    private static OccurrenceInfo getPreviousOccurrence(TimeWindowFilterSettings settings, ZonedDateTime now) {
+        ZonedDateTime start = settings.getStart();
+        OccurrenceInfo emptyOccurrence = new OccurrenceInfo();
+        if (now.isBefore(start)) {
+            return emptyOccurrence;
+        }
+
+        final RecurrencePatternType patternType = settings.getRecurrence().getPattern().getType();
+        OccurrenceInfo occurrenceInfo;
+        if (patternType == RecurrencePatternType.DAILY) {
+            occurrenceInfo = getDailyPreviousOccurrence(settings, now);
+        } else {
+            occurrenceInfo = getWeeklyPreviousOccurrence(settings, now);
+        }
+
+        final RecurrenceRange range = settings.getRecurrence().getRange();
+        if (range.getType() == RecurrenceRangeType.END_DATE
+            && occurrenceInfo.previousOccurrence != null
+            && occurrenceInfo.previousOccurrence.isAfter(range.getEndDate())) {
+            return emptyOccurrence;
+        }
+        if (range.getType() == RecurrenceRangeType.NUMBERED
+            && occurrenceInfo.numberOfOccurrences > range.getNumberOfRecurrences()) {
+            return emptyOccurrence;
+        }
+
+        return occurrenceInfo;
+    }
+
+    /**
+     * Find the closest previous recurrence occurrence before the provided time stamp according to the "Daily" recurrence pattern.
+     *
+     * @return The return result contains two property, one is previousOccurrence, the other is numberOfOccurrences.
+     * previousOccurrence: The closest previous occurrence.
+     * numberOfOccurrences: The number of complete recurrence intervals which have occurred between the time and the recurrence start.
+     */
+    private static OccurrenceInfo getDailyPreviousOccurrence(TimeWindowFilterSettings settings, ZonedDateTime now) {
+        final ZonedDateTime start = settings.getStart();
```

Comment:
I know that we've already got this in `getPreviousOccurrence` method

``` java
if (now.isBefore(start)) {
            return emptyOccurrence;
        }
```
Can we also set a get in this method to prevent now < start which will cause negative numberOfOccurrences

Reply:
I don't think we need this check here. 
`getDailyPreviousOccurrence` is only called when now is after start.